src/containers/EditProfileContainer.js

```diff
@@ -22,6 +22,8 @@ const EditProfileContainer = props => {
     myUserInfo,
   } = props
 
+  if (!myUserInfo.profileReady) history.push('/registration/info')
```

Comment:
The only thing I noticed is, that if user refreshes the page in the middle registration flow, it seems to continue from the same page, but actually profile is not ready after completing the flow, so user is redirected to registration and has to do it "twice". I don't think this will be a common scenario or a big problem, but just something to consider if we could improve it somehow. Maybe we could skip the steps that have been filled already or something.

Reply:
You mean when first filling the registration and then the user refreshes, some of the previously filled in data is lost? Could that be the problem for the users who didn't get their nickname shown right.. That should probably be prevented somehow.

Reply:
Hmm I'm not completely sure where the problem exactly is. Looking into it not :) Probably data is not lost, but maybe some async problem with redirecting and profileReady being updated after that or something.

Reply:
So I added some async/await stuff for onClick action in StepButtons. I think it works now as intended and even if user refreshes the page in the middle of registration flow. Maybe you could verify this? :)

Reply:
I tried and couldn't really recreate the problem, for me it saved all the info in user's profile even if I did update (twice) in the middle of registration flow. Adding the awaits sounds like a good idea, though. I think this is good to go now?